queries.sql

```diff
@@ -1,8 +1,19 @@
-SELECT * FROM animals WHERE name LIKE '%mon';
-SELECT name FROM animals WHERE date_of_birth BETWEEN '2016/01/01' AND '2019/12/31';
-SELECT name FROM animals WHERE neutered=TRUE AND escape_attempts < 3;
-SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
-SELECT date_of_birth, escape_attempts FROM animals WHERE weight_kg > 10.5;
-SELECT * FROM animals WHERE neutered = TRUE;
-SELECT * FROM animals WHERE NOT name = 'Gabumon';
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3 
+
+SELECT A.name, V.name, Vi.visit_date AS vdate FROM vets V JOIN visits Vi ON V.id = Vi.vets_id JOIN animals A ON Vi.animals_id = A.id WHERE V.name = 'William Tatcher' ORDER BY vdate DESC LIMIT 1;
+
+SELECT V.name, COUNT(S.species_id) FROM vets V JOIN specialization S ON V.id = S.vets_id GROUP BY V.name HAVING V.name = 'Stephanie Mendez';
+
+SELECT V.name, S.name FROM vets V FULL JOIN specialization SV ON V.id = SV.vets_id LEFT JOIN species S on SV.species_id = S.id GROUP BY V.name, S.name;
+
+SELECT  V.name, A.name, Vi.visit_date FROM animals A JOIN visits Vi ON A.id = Vi.animals_id JOIN vets V ON V.id = Vi.vets_id WHERE V.name = 'Stephanie Mendez' AND  Vi.visit_date BETWEEN '2020-04-01' AND '2020-08-30'; 
+
+WITH maxt AS (SELECT A.name AS ani_name, Count(Vi.animals_id) AS count, Max(Count(Vi.animals_id)) OVER() AS maxcount FROM animals A JOIN visits Vi ON A.id = Vi.animals_id GROUP  BY A.name) SELECT ani_name, count
+FROM maxt WHERE  count = maxcount; 
+
+SELECT A.name, V.name, Vi.visit_date FROM animals A JOIN visits Vi ON A.id = Vi.animals_id JOIN vets V ON V.id = Vi.vets_id WHERE  V.name = 'Maisy Smith ' ORDER  BY Vi.visit_date LIMIT  1; 
+
+SELECT Vi.visit_date, A.name AS animal_name, A.date_of_birth, A.escape_attempts, A.neutered, A.weight_kg, species.name AS specie, V.name AS vet, V.age, V.date_of_graduation FROM species JOIN animals A ON species.id = A.species_id JOIN visits Vi ON A.id = Vi.animals_id JOIN vets V ON Vi.vets_id = V.id ORDER  BY Vi.visit_date DESC LIMIT  1; 
+
+WITH view AS (SELECT S.name, V.name AS vet FROM vets V FULL JOIN visits Vi ON  V.id = Vi.vets_id FULL JOIN specialization SV ON SV.vets_id = V.id FULL JOIN species S ON S.id = SV.species_id WHERE S.name is NULL) SELECT vet, COUNT(vet) FROM view GROUP BY vet;
+
```

Comment:
- [OPTIONAL] How many times visits where with a vet that did not specialize in that animal's species? You could use this approach 👇🏻 

```sql
  SELECT count(*)
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN vets ON vets.id = visits.vet_id
  WHERE animals.species_id NOT IN (SELECT species_id FROM specializations WHERE vet_id = vets.id)
```

Reply:
I have used this query. it works well. thank you.